derive-artifact-version.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+#
+# Copyright 2017, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# (c) 2017, Jesse Pretorius <jesse.pretorius@rackspace.co.uk>
+
+"""Determine the version string to use for artifact publishing."""
+
+from __future__ import print_function
+
+import re
+import subprocess
+
+GIT_FOLDER =(
+    '/opt/rpc-openstack'
+)
+CMD_GET_TAG =(
+    "git describe --tags --abbrev=0"
+)
+CMD_GET_BRANCH =(
+    "git branch --contains $(git rev-parse HEAD) | grep ^\* | sed 's/^\* //'"
+)
+
+def main():
+    """Run the main application."""
+
+    current_tag = subprocess.check_output(
+                      CMD_GET_TAG, cwd=GIT_FOLDER, shell=True
+                  ).strip(' \n\t')
```

Comment:
Does 'whitespace' include line breaks and carriage returns? I specifically want to remove all of those things.